providers/base/bin/gateway_ping_test.py

```diff
@@ -219,85 +219,125 @@ def ping(host, interface, count, deadline, verbose=False):
+    )
     parser.add_argument(
-        "-v", "--verbose", action='store_true', help=_("be verbose"))
+        "-v", "--verbose", action="store_true", help=_("be verbose")
+    )
     parser.add_argument(
-        "-I", "--interface", help=_("use specified interface to send packets"))
-    args = parser.parse_args()
+        "-I", "--interface", help=_("use specified interface to send packets")
+    )
+    args = parser.parse_args(argv)
     # Ensure count and deadline make sense. Adjust them if not.
     if args.deadline != default_delay and args.count != default_count:
         # Ensure they're both consistent, and exit with a warning if not,
         # rather than modifying what the user explicitly set.
         if args.deadline <= args.count:
             # FIXME: this cannot ever be translated correctly
-            print(_(
-                "ERROR: not enough time for {0} pings in {1} seconds"
-            ).format(args.count, args.deadline))
-            return 1
+            raise SystemExit(
+                _(
+                    "ERROR: not enough time for {0} pings in {1} seconds"
+                ).format(args.count, args.deadline)
+            )
     elif args.deadline != default_delay:
         # Adjust count according to delay.
         args.count = args.deadline - 1
         if args.count < 1:
             args.count = 1
         if args.verbose:
             # FIXME: this cannot ever be translated correctly
-            print(_(
-                "Adjusting ping count to {0} to fit in {1}-second deadline"
-            ).format(args.count, args.deadline))
+            print(
+                _(
+                    "Adjusting ping count to {0} to fit in {1}-second deadline"
+                ).format(args.count, args.deadline)
+            )
     else:
         # Adjust delay according to count
         args.deadline = args.count + 1
         if args.verbose:
             # FIXME: this cannot ever be translated correctly
-            print(_(
-                "Adjusting deadline to {0} seconds to fit {1} pings"
-            ).format(args.deadline, args.count))
+            print(
+                _("Adjusting deadline to {0} seconds to fit {1} pings").format(
+                    args.deadline, args.count
+                )
+            )
+    return args
+
+
+def main(argv) -> int:
+    gettext.textdomain("com.canonical.certification.checkbox")
+    gettext.bindtextdomain(
+        "com.canonical.certification.checkbox",
+        os.getenv("CHECKBOX_PROVIDER_LOCALE_DIR", None),
+    )
+
+    args = parse_args(argv)
+
     # If given host is not pingable, override with something pingable.
     host = get_host_to_ping(
-        interface=args.interface, verbose=args.verbose, default=args.host)
+        interface=args.interface, verbose=args.verbose, default=args.host
+    )
     if args.verbose:
         print(_("Checking connectivity to {0}").format(host))
-    ping_summary = None
```

Comment:
Would't it be better to have something like:
```
    if host:
        ping_summary = ping(
            host, args.interface, args.count, args.deadline, args.verbose
        )
    else:
        ping_summary = {"received": 0, "cause": "Could not find a host"}

    if ping_summary["received"] == 0:
        print(_("No Internet connection"))
        if ping_summary.get("cause"):
            print("Possible cause: {}".format(ping_summary["cause"]))
        return 1
```

Reply:
good idea